### DISCUSSION TITLE
Make `useRouteError` expose the `info` param from React's error boundaries

### DISCUSSION THREAD

**Discussion by mark-dr** [remix-run/react-router#11206] [Proposals]
When a React error boundary catches an error with [`componentDidCatch`](https://react.dev/reference/react/Component#componentdidcatch), it receives two pieces of information:

> `error`: The error that was thrown. In practice, it will usually be an instance of [`Error`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error) but this is not guaranteed because JavaScript allows to [`throw`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw) any value, including strings or even `null`.
> `info`: An object containing additional information about the error. Its `componentStack` field contains a stack trace with the component that threw, as well as the names and source locations of all its parent components. In production, the component names will be minified. If you set up production error reporting, you can decode the component stack using sourcemaps the same way as you would do for regular JavaScript error stacks.

In a `react-router`, `useRouteError` provides access to the `error`, but I can't see a way to access the `info`. I've therefore reluctantly voted for https://github.com/remix-run/react-router/discussions/10166 as a way to eject from react-router's error catching, but I'd retract that vote if `react-router` could give me the `info` object.

Possible solutions, trying to avoid breaking changes:

- Add a new hook (`useRouteReactErrorInfo`)
- Add a new parameter (e.g. `useRouteError({ includeReactErrorInfo: true })`) that makes `useRouteError` return structured information when the error is from a React error boundary
- Append the `info` to the `error` Object as it passes through the React error boundary
- Goofy option: Make `useRouteError` return the `error` on the first time it's called within a render, and `info` (or nothing, for non-rendering errors) on the second

EDIT: On reflection, for our use case, we don't actually need the data for rendering (just for logging), so a `RouterProvider`-level callback would also work great.

**Comment by brophdawg11** [comment]
Closing this as a duplicate of #9881
